### PR TITLE
feat(lh-86965): add msp_managed_tenant data source

### PR DESCRIPTION
**Pull request by siddhuwarrier**
https://jira-eng-rtp3.cisco.com/jira/browse/LH-86965

### Description

Add a datasource to read an MSP managed tenant

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -282,3 +282,7 @@ func (c *Client) CreateTenantUsingMspPortal(ctx context.Context, createInput ten
 func (c *Client) ReadMspManagedTenant(ctx context.Context, readByUidInput tenants.ReadByUidInput) (*tenants.MspTenantOutput, error) {
```
Comment (talhazi) [inline client/client.go:282]: Maybe change it to `ReadMspManagedTenantByUid` to clarify the difference?

File: client/internal/url/url.go [hunk 2]
```diff
@@ -213,3 +213,7 @@ func CreateMspManagedTenant(baseUrl string) string {
 func ReadMspManagedTenant(baseUrl string, tenantUid string) string {
```
Comment (talhazi) [inline client/internal/url/url.go:213]: same

File: provider/examples/data-sources/msp/tenants/api_token.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+Paste your API token here
```
Comment (talhazi) [inline provider/examples/data-sources/msp/tenants/api_token.txt:1]: Nice!

**Review by talhazi — APPROVED** [review]
LGTM!

File: client/client.go (hunk 1 above)
Comment (siddhuwarrier) [reply client/client.go:282]: Done

File: client/internal/url/url.go (hunk 2 above)
Comment (siddhuwarrier) [reply client/internal/url/url.go:213]: Done

**Pull request merged**